apps/address-service/domains/address/schema/AddressSource.test.js

```diff
@@ -0,0 +1,217 @@
+/**
+ * Generated by `createschema address.AddressSource 'source:Text;'`
+ */
+
+const {
+    makeLoggedInAdminClient,
+    makeClient,
+    UUID_RE,
+    DATETIME_RE,
+    expectToThrowGQLError,
+    catchErrorFrom,
+} = require('@open-condo/keystone/test.utils')
+
+const {
+    expectToThrowAuthenticationErrorToObj,
+    expectToThrowAuthenticationErrorToObjects,
+    expectToThrowAccessDeniedErrorToObj,
+    expectToThrowAccessDeniedErrorToObjects,
+} = require('@open-condo/keystone/test.utils')
+
+const {
+    makeClientWithNewRegisteredAndLoggedInUser,
+    makeClientWithSupportUser,
+} = require('@address-service/domains/user/utils/testSchema')
+
+const {
+    AddressSource,
+    createTestAddressSource,
+    updateTestAddressSource,
+} = require('@address-service/domains/address/utils/testSchema')
+
+const faker = require('faker')
+
+let adminClient, supportClient, userClient, anonymousClient
+
+describe('AddressSource', () => {
+    beforeAll(async () => {
+        adminClient = await makeLoggedInAdminClient()
+        supportClient = await makeClientWithSupportUser()
+        userClient = await makeClientWithNewRegisteredAndLoggedInUser()
+        anonymousClient = await makeClient()
+    })
+
+    describe('CRUD tests', () => {
+        describe('create', () => {
+            test('admin can', async () => {
+                const [obj, attrs] = await createTestAddressSource(adminClient)
+
+                expect(obj.id).toMatch(UUID_RE)
+                expect(obj.dv).toEqual(1)
+                expect(obj.sender).toEqual(attrs.sender)
+                expect(obj.v).toEqual(1)
+                expect(obj.newId).toEqual(null)
+                expect(obj.deletedAt).toEqual(null)
+                expect(obj.createdBy).toEqual(expect.objectContaining({ id: adminClient.user.id }))
+                expect(obj.updatedBy).toEqual(expect.objectContaining({ id: adminClient.user.id }))
+                expect(obj.createdAt).toMatch(DATETIME_RE)
+                expect(obj.updatedAt).toMatch(DATETIME_RE)
+            })
+
+            test('support can\'t', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await createTestAddressSource(supportClient)
+                })
+            })
+
+            test('user can\'t', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await createTestAddressSource(userClient)
+                })
+            })
+
+            test('anonymous can\'t', async () => {
+                await expectToThrowAuthenticationErrorToObj(async () => {
+                    await createTestAddressSource(anonymousClient)
+                })
+            })
+
+            test('Can not create two sources with the same address', async () => {
```

Comment:
Could you please write a similar test for 'update' operation

Reply:
Done